### PR TITLE
Add cache manager module to clean up the generated code cache

### PR DESCRIPTION
**Pull request by egparedes**
## Description

Add cache manager module to clean up and check the config of the generated code cache

### CODE REVIEW

**Review by tehrengruber — APPROVED** [review]
Looks good.

File: src/gt4py/gt_cache_manager.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        print("\nRoot paths:")
+        print(f"\tconfig = {_get_root()})")
+        print(f"\targs = {args.root})")
+        caches_list = "\n\t".join(str(c) for c in caches)
+        print(f"\nFound:\n\t{caches_list}")
+
+    else:
+        raise AssertionError(f"command={args.command}")
+
+    print()
```
Comment (gronerl) [inline src/gt4py/gt_cache_manager.py:88]: `args.command == "clean"` case ends with `\n`, `args.command == "status"` doesn't. the gt_src_manager does not end with the `print()`

**Review by gronerl — APPROVED** [review]
It looks sound, just merge if it blocks you.

But wouldn't it make sense if the cleaning from the tests, and the finding the caches when compiling stencils would also use these utilities here? Otherwise we have two code paths doing the same thing, also it would at least mean that this code is called in the tests.

**Comment by egparedes** [comment]
> But wouldn't it make sense if the cleaning from the tests, 
That's exactly the point why I want this merged. The change in the tests configs is coming in the next PR.

**Pull request merged**